.circleci/config.yml

```diff
@@ -177,9 +229,76 @@ jobs:
           path: /tmp/resource-usage
           destination: resource-usage
 
+  unit-test-windows:
+    executor:
+      name: win/default
+      shell: bash.exe
```

Comment:
why do we default to bash.exe but then override and use powershell.exe in many of the following?

Reply:
The codecov/upload command does not work if we default to powershell.exe as it internally uses a bash script. I tried to find whether it can be used with powershell.exe as the default shell, but there is no documentation for Windows. There are Github issues on their repo regarding the use of Windows but couldn't find anything concrete from that. 
It is also not possible to specify the shell for the codecov/upload command. It is run in the default shell of the job. So the easiest solution was to make bash.exe the default shell.

Reply:
I see that you removed `shell: bash.exe`, and the download codecov uploader step in CircleCI is failing because of it. 
Should I make the default shell bash again?

Reply:
Thanks for the explanation and sorry for breaking this with my commit. Yes, please do revert that part of my change. Can you add a comment explaining why we're defaulting to `bash` as I expect this to be confusing in the future?

Reply:
Added a comment in the config.